apps/swirl-docs/src/components/ComponentPreview/CodePreview.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  FlipIconCheckStrong,
+  FlipIconCopy,
+  FlipIconExpandLess,
+  FlipIconExpandMore,
+  FlipIconOpenInNew,
+} from "@getflip/swirl-components-react";
+import { FunctionComponent, useCallback, useEffect, useState } from "react";
+import {
+  Prop,
+  SwirlComponentCodePreview,
+} from "@swirl/lib/components/src/components.model";
+import { CodeSandboxButton } from "./CodeSandboxButton";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import classNames from "classnames";
+import Link from "next/link";
+import SyntaxHighlighter from "react-syntax-highlighter";
+
+import dark from "react-syntax-highlighter/dist/cjs/styles/prism/a11y-dark";
+import NoSsr from "../Layout/NoSsr";
+import prettier from "prettier/standalone";
+import prettierHTML from "prettier/parser-html";
+
+interface CodePreviewProps {
+  component: SwirlComponentCodePreview | null;
+}
+
+export const CodePreview: FunctionComponent<CodePreviewProps> = ({
+  component,
+}) => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+  const [code, setCode] = useState<string>("");
+
+  const generateCode = useCallback(() => {
```

Comment:
**suggestion:** Could we re-generate the code when another variant is selected?

Reply:
I implemented the basic setup for the logic. A open to do is to be able to send queried data to storybook through url params based on the props. But the props, example logic is a seperate ticket and out of scope. i will create a ticket for that

Reply:
e. created a ticket for that: [APP-12682](https://flipapp.myjetbrains.com/youtrack/issue/APP-12682) Props to Storybook-url-params parser